### PR TITLE
feat(like): add like endpoint, sync with algolia

### PR DESCRIPTION
**Pull request by sterlingrules**
Supports https://github.com/SoftwareEngineeringDaily/sedaily-front-end/issues/624

### CODE REVIEW

File: server/controllers/like.controller.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *         $ref: '#/responses/Unauthorized'
+ *       '404':
+ *         $ref: '#/responses/NotFound'
+ */
+async function likePost(req, res) {
+  const { postId } = req.params;
+  const { _id: userId } = req.user;
+  const query = { postId, userId };
+  const like = new Like(query);
+  const likeActive = await Like.findOne(query, (err, reply) => Promise.resolve(reply));
```
Comment (marcosschlup) [inline server/controllers/like.controller.js]: Is this `Promise.resolve` needed? What's the idea here?

File: server/models/like.model.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ *      dateCreated:
+ *        type: string
+ *        format: date-time
+ *        description: Date comment created
+ *        example: 2017-10-30T01:05:39.674Z
+ */
+const LikeSchema = new mongoose.Schema({
+  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
+  postId: { type: mongoose.Schema.Types.ObjectId, ref: 'Post' },
+});
```
Comment (marcosschlup) [inline server/models/like.model.js]: Missing `createdAt` ? 
Other entities have a field named dateCreated. It seems to be the most repeated. I found a createdAt in users. In fact, I don't know if we can conclude that there is a pattern. It's up to you

File: server/controllers/like.controller.js (hunk 1 above)
Comment (sterlingrules) [reply server/controllers/like.controller.js]: Ah, thanks. Yea, that's not needed.

File: server/models/like.model.js (hunk 2 above)
Comment (sterlingrules) [reply server/models/like.model.js]: Yea, I'm use to `createdAt` as well, but I've been seeing `dateCreated` throughout the codebase. I was trying to match how votes are handled and that model didn't have a creation property so I didn't add it initially. I'll go ahead and add it.

**Pull request merged**